massa-client/src/settings.rs

```diff
@@ -1,28 +1,54 @@
 // Copyright (c) 2022 MASSA LABS <info@massa.net>
 
 //! Build here the default client settings from the configuration file toml
+#[cfg(not(feature = "testing"))]
 use massa_models::constants::build_massa_settings;
 use massa_time::MassaTime;
 use serde::Deserialize;
-use std::{net::IpAddr, path::PathBuf};
+use std::{net::SocketAddr, path::PathBuf};
 
+#[cfg(feature = "testing")]
+lazy_static::lazy_static! {
+    pub static ref SETTINGS: Settings = Default::default();
+}
+
+#[cfg(not(feature = "testing"))]
 lazy_static::lazy_static! {
     pub static ref SETTINGS: Settings = build_massa_settings("massa-client", "MASSA_CLIENT");
 }
 
 #[derive(Debug, Deserialize, Clone)]
 pub struct Settings {
-    pub default_node: DefaultNode,
+    pub default_node: NodeAddr,
```

Comment:
This change cause this error : 
```rust
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: missing field `public`', /Users/aurelienfoucault/massa/massa-models/src/node_configuration/massa_settings.rs:78:25
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I think you should change the format in the config file.

Reply:
You're right!